src/charm.py

```diff
@@ -195,15 +214,28 @@ def _on_start(self, event: StartEvent) -> None:
             )
             event.defer()
 
-    def _on_update_status(self, _):
+    def _on_upgrade(self, event) -> None:
+        container = self.unit.get_container(Config.CONTAINER_NAME)
+        try:
+            self._configure_layers(container=container)
+        except ContainerNotReadyError:
+            self.status.set_and_share_status(Config.Status.UNHEALTHY_UPGRADE)
+            logger.error("Failed to replan")
+            event.defer()
+            return
+
+        self.status.set_and_share_status(ActiveStatus())
+        self.upgrade._reconcile_upgrade(event, during_upgrade=True)
+        if self.upgrade._upgrade.is_compatible:
```

Comment:
should something be done if the upgrade isn't compatible? - or is that handled by the fact that we never set the status of the upgrade to HEALTHY

Reply:
If the upgrade isn't compatible, there will be a status saying that (set by `_reconcile_upgrade`) and we can either rollback or force the upgrade :)

Reply:
thank you <3